internal/storage/postgres/tree/trees_test.go

```diff
@@ -0,0 +1,225 @@
+		assert.Error(t, err)
+		assert.Nil(t, deletedTree)
+
+		for _, img := range images {
+			_, err = suite.Store.GetImageByID(context.Background(), img.ID)
+			assert.Error(t, err)
+		}
+	})
+
+	t.Run("should return error if tree not found", func(t *testing.T) {
+		// given
+		suite.ResetDB(t)
+		r := NewTreeRepository(suite.Store, mappers)
+
+		// when
+		err := r.Delete(context.Background(), 99)
+
+		// then
+		assert.Error(t, err)
+	})
+
+	t.Run("should return error when context is canceled", func(t *testing.T) {
+		// given
+		r := NewTreeRepository(suite.Store, mappers)
+		ctx, cancel := context.WithCancel(context.Background())
+		cancel()
+
+		// when
+		err := r.Delete(ctx, 1)
+
+		// then
+		assert.Error(t, err)
+	})
+}
+
+func TestTreeRepository_DeleteAndUnlinkImages(t *testing.T) {
+	t.Run("should delete tree and unlink all images successfully", func(t *testing.T) {
+		// given
+		suite.ResetDB(t)
+		suite.InsertSeed(t, "internal/storage/postgres/seed/test/tree")
+		r := NewTreeRepository(suite.Store, mappers)
+		treeID := int32(1)
+		sqlImages, err := suite.Store.GetAllImagesByTreeID(context.Background(), treeID)
+		if err != nil {
+			t.Fatal(err)
+		}
+		images := mappers.iMapper.FromSqlList(sqlImages)
+		assert.NotEmpty(t, images)
+
+		// when
+		err = r.DeleteAndUnlinkImages(context.Background(), treeID)
+		assert.NoError(t, err)
+
+		deletedTree, err := r.GetByID(context.Background(), treeID)
+
+		// then
+		assert.Error(t, err)
+		assert.Nil(t, deletedTree)
+
+		for _, img := range images {
+			_, err = suite.Store.GetImageByID(context.Background(), img.ID)
+			assert.NoError(t, err)
+		}
+	})
+
+	t.Run("should return error by canceling the context", func(t *testing.T) {
+		// given
+		suite.ResetDB(t)
+		r := NewTreeRepository(suite.Store, mappers)
+		ctx, cancel := context.WithCancel(context.Background())
+		cancel()
+
+		// when
+		err := r.DeleteAndUnlinkImages(ctx, 1)
+
+		// then
+		assert.Error(t, err)
+	})
+}
+func TestTreeRepository_UnlinkImage(t *testing.T) {
+	t.Run("should unlink a specific image from tree successfully", func(t *testing.T) {
+		// given
+		suite.ResetDB(t)
+		suite.InsertSeed(t, "internal/storage/postgres/seed/test/tree")
+		r := NewTreeRepository(suite.Store, mappers)
+		treeID := int32(1)
+		imageID := int32(1)
+
+		isContained, err := isContainedInTreeImages(imageID, treeID)
+		assert.NoError(t, err)
+		assert.True(t, isContained)
+
+		// when
+		err = r.UnlinkImage(context.Background(), treeID, imageID)
+
+		// then
+		assert.NoError(t, err)
+		isContained, err = isContainedInTreeImages(imageID, treeID)
+		assert.NoError(t, err)
+		assert.False(t, isContained)
+	})
+
+	t.Run("should return error if image or tree not found", func(t *testing.T) {
+		// given
+		suite.ResetDB(t)
+		r := NewTreeRepository(suite.Store, mappers)
+
+		// when
+		err := r.UnlinkImage(context.Background(), 99, 99)
+
+		// then
+		assert.Error(t, err)
+	})
+}
+
+func TestTreeRepository_UnlinkTreeClusterID(t *testing.T) {
```

Comment:
- check for invalid id - 0 and -1
- check if error on not found.
- check for not set treecluster